### PR TITLE
feat: add artist to auctionResultsByFollowedArtists query

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [CX-1616]

## Description

Adds an artist field to auction results to be able to fetch information about the artist for the auction **results by followed artists** query.  

Though the field will be available everywhere on `AuctionResult`, just the `AuctionResultsByFollowedArtists` query will provide the data for it.  

[CX-1616]: https://artsyproduct.atlassian.net/browse/CX-1616

### CODE REVIEW

File: src/schema/v2/me/auction_results_by_followed_artists.ts [hunk 1]
```diff
@@ -103,6 +102,16 @@ const AuctionResultsByFollowedArtists: GraphQLFieldConfig<
...
           const totalPages = Math.ceil(total_count / size)
+
+          // enrich result with artist data
+          const items = _embedded.items.map((auctionResult) => {
+            const artist = followedArtists.find(
+              (artist) => artist.artist._id == auctionResult.artist_id
+            )
+            auctionResult.artist = artist?.artist
+            return auctionResult
+          })
```
Comment (olerichter00) [inline src/schema/v2/me/auction_results_by_followed_artists.ts]: Note: this doesn't fetch any additional data and just maps the artists from Gravity to the auction results by `artist id`. We could check if the artist field is requested but I guess mapping the data shouldn't slow the query down very much.

File: src/schema/v2/me/auction_results_by_followed_artists.ts [hunk 2]
```diff
@@ -103,6 +102,16 @@ const AuctionResultsByFollowedArtists: GraphQLFieldConfig<
...
           const totalPages = Math.ceil(total_count / size)
+
+          // enrich result with artist data
+          const items = _embedded.items.map((auctionResult) => {
+            const artist = followedArtists.find(
+              (artist) => artist.artist?._id == auctionResult.artist_id
+            )
+            auctionResult.artist = artist.artist
+            return auctionResult
+          })
```
Comment (olerichter00) [inline src/schema/v2/me/auction_results_by_followed_artists.ts:113]: Note: this doesn't fetch any additional data and just maps the artists from Gravity to the auction results by artist id. We could check if the artist field is requested but I guess mapping the data shouldn't slow the query down very much.

**Review by JenqaaaBY — APPROVED** [review]
Looks good to me

**Review by brainbicycle — APPROVED** [review]
Nice!

**Pull request merged**